src/main/java/org/apache/commons/lang3/text/translate/JavaUnicodeEscaper.java

```diff
@@ -110,7 +112,8 @@ public JavaUnicodeEscaper(final int below, final int above, final boolean betwee
      */
     @Override
     protected String toUtf16Escape(final int codepoint) {
-        final char[] surrogatePair = Character.toChars(codepoint);
+        @SuppressWarnings("assignment.type.incompatible") // array has @MinLen(2) if codepoint > 0xffff, which is checked when this function is called
+        final char @MinLen(2) [] surrogatePair = Character.toChars(codepoint);
```

Comment:
Is it possible to annotate `codepoint` with an `@IntRange` annotation that expresses that information?

That won't prevent the warning suppression, but it will require that the check happens at call-sites.